build_tools/aws-sdk-code-generator/templates/endpoints_module.mustache

```diff
@@ -11,19 +11,21 @@ module {{module_name}}
     {{#endpoint_classes}}
     class {{name}}
       def self.build(context)
-        {{module_name}}::EndpointParameters.new(
+        {{module_name}}::EndpointParameters.create(
+          context.config,
           {{#parameters}}
-          {{#static_string?}}
-          {{key}}: "{{{value}}}",
-          {{/static_string?}}
-          {{^static_string?}}
           {{key}}: {{{value}}},
-          {{/static_string?}}
           {{/parameters}}
         )
       end
     end
 
     {{/endpoint_classes}}
+
+    OPERATION_PARAM_MAP = {
```

Comment:
Why this over using the method with a case statement?

Reply:
We know have classes only for certain operations (ones which need operation specific values).  I wanted to keep that knowledge isolated here rather than duplicating it in the endpoints module.

Reply:
That's fine